internal/handlers/metrics/metrics.go

```diff
@@ -0,0 +1,309 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/timuraipov/alert/internal/config"
+	"github.com/timuraipov/alert/internal/domain/metric"
+	"github.com/timuraipov/alert/internal/filestorage"
+	"github.com/timuraipov/alert/internal/logger"
+	"go.uber.org/zap"
+)
+
+type MetricStorage interface {
+	Save(metric metric.Metrics) (metric.Metrics, error)
+	GetAll() []metric.Metrics
+	GetByTypeAndName(metricType, metricName string) (metric.Metrics, bool)
+}
+type MetricHandler struct {
+	Storage         MetricStorage
+	fileStorage     *filestorage.Storage
+	config          *config.Config
+	IsNeedSyncFlush bool
+}
+
+func New(storage MetricStorage, fileStorage *filestorage.Storage, cfg *config.Config) *MetricHandler {
+	metricsHandler := &MetricHandler{
+		Storage:     storage,
+		fileStorage: fileStorage,
+		config:      cfg,
+	}
+	metricsHandler.init()
+	return metricsHandler
+}
+func (mh *MetricHandler) init() {
+	if mh.config.Restore {
+		mh.load()
+	}
+	if mh.config.StoreInterval == 0 {
+		mh.IsNeedSyncFlush = true
+	} else {
```

Comment:
```suggestion
	} else if mh.config.StoreInterval > 0 {
```
Хотя эту проверку лучше в разбор конфига вытащить. И если зачем-то передано отрицательное число - падать.